### PR TITLE
OCPBUGS-3405: Redact pull secret from agent-gather

### PR DESCRIPTION
**Pull request by zaneb**
The agent-gather output should not contain the user's pull secret. Redact it from `/etc/assisted/manifests/pull-secret.yaml`.

### CODE REVIEW

File: pkg/asset/agent/manifests/agentpullsecret.go [hunk 1]
```diff
@@ -118,17 +113,40 @@ func (a *AgentPullSecret) finish() error {
 		return errors.Wrapf(err, "invalid PullSecret configuration")
 	}
 
+	// Normalise the JSON formatting so that we can redact the file reliably
```
Comment (andfasano) [inline pkg/asset/agent/manifests/agentpullsecret.go:116]: nit: would be nice to move this block into its own method, for readability

File: data/data/agent/files/usr/local/bin/agent-gather [hunk 2]
```diff
@@ -10,6 +10,10 @@ function gather_agent_data() {
 	( >&2 echo -n "Gathering agent installation data" )
 	mkdir -p "${ARTIFACTS_DIR}/etc"
 	cp -a /etc/assisted{,-service} "${ARTIFACTS_DIR}/etc/"
+        # Redact pull secrets
+        for manifest in "${ARTIFACTS_DIR}"/etc/assisted/manifests/*; do
+            sed -i -e '/"auth":/ s/: *"[A-Za-z0-9+/]*=*"/: "<redacted>"/g' "${manifest}"
+        done
```
Comment (andfasano) [inline data/data/agent/files/usr/local/bin/agent-gather:16]: What about the journals? They could also potentially capture confidential data (secrets included)

Comment (zaneb) [reply data/data/agent/files/usr/local/bin/agent-gather:16]: They shouldn't, but it's hard to know how to stop them if they did.

Comment (zaneb) [reply data/data/agent/files/usr/local/bin/agent-gather]: OK, I figured out a way to do it.

**Review by pawanpinjarkar — COMMENTED** [review]
/lgtm

**Comment by zaneb** [comment]
ci/prow/okd-scos-images

**Pull request merged**